File: B3WM.Shared/Models/StructureStorageItem.cs
using B3WM.Shared.Interfaces;

namespace B3WM.Shared.Models
{
    public class StructureStorageItem : ICloneable
    {
        public DateTime Date { get; set; }

        public string Symbol { get; set; }

        //InMinutes
        public int TimeFrame { get; set; }

        public double UpBorder { get; set; }

        public double DownBorder { get; set; }

        public double UpAuxBorder { get; set; }

        public double DownAuxBorder { get; set; }

        public object Clone()
        {
            var json = System.Text.Json.JsonSerializer.Serialize(this);
            return System.Text.Json.JsonSerializer.Deserialize<StructureStorageItem>(json) ?? new StructureStorageItem();
        }

        public override string ToString()
        {
            return $"Date: {Date.ToString("dd/MM/yyyy HH:mm:ss")} | " +
                $"UpBorder:{UpBorder.ToString("00.00")} | " +
                $"UpAuxBorder:{UpAuxBorder.ToString("00.00")} | " +
                $"DownBorder:{DownBorder.ToString("00.00")} | " +
                $"DownAuxBorder:{DownAuxBorder.ToString("00.00")}";
        }
    }
}

File: B3WM.Flutter/lib/models/structure_storage_item.dart
class StructureStorageItem {
  final DateTime date;
  final String symbol;
  final int timeFrame;
  final double upBorder;
  final double downBorder;
  final double upAuxBorder;
  final double downAuxBorder;

  StructureStorageItem({
    required this.date,
    required this.symbol,
    required this.timeFrame,
    required this.upBorder,
    required this.downBorder,
    required this.upAuxBorder,
    required this.downAuxBorder,
  });

  factory StructureStorageItem.fromJson(Map<String, dynamic> json) {
    return StructureStorageItem(
      date: DateTime.parse(json['date'] as String),
      symbol: json['symbol'] as String,
      timeFrame: json['timeFrame'] as int,
      upBorder: (json['upBorder'] as num).toDouble(),
      downBorder: (json['downBorder'] as num).toDouble(),
      upAuxBorder: (json['upAuxBorder'] as num).toDouble(),
      downAuxBorder: (json['downAuxBorder'] as num).toDouble(),
    );
  }

  Map<String, dynamic> toJson() => {
        'date': date.toIso8601String(),
        'symbol': symbol,
        'timeFrame': timeFrame,
        'upBorder': upBorder,
        'downBorder': downBorder,
        'upAuxBorder': upAuxBorder,
        'downAuxBorder': downAuxBorder,
      };
}
